docs/source/user_guide/pipelines.md

```diff
@@ -88,9 +88,14 @@ The [tutorials](/getting_started/tutorials.md) provide comprehensive step-by-ste
          - **Environment variables**
            - A list of environment variables to be set in the container that executes the Jupyter notebook or script. Format: `ENV_VAR_NAME=value`. Entries that are empty (`ENV_VAR_NAME=`) or malformed are ignored.
          - **Data volume**
-           - A list of [Persistent Volume Claims](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) (PVC) to be mounted into the container that executes the Jupyter notebook or script. Format: `/mnt/path=existing-pvc-name`. Entries that are empty (`/mnt/path=`) or malformed are ignored. 
+           - A list of [Persistent Volume Claims](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) (PVC) to be mounted into the container that executes the Jupyter notebook or script. Format: `/mnt/path=existing-pvc-name`. Entries that are empty (`/mnt/path=`) or malformed are ignored. Entries with a PVC name considered to be an [invalid Kubernetes resource name](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names) will raise a validation error after pipeline submission or export.
            - The referenced PVCs must exist in the Kubernetes namespace where the generic pipeline nodes are executed.
            - Data volumes are not mounted when the pipeline is executed locally.
+         - **Kubernetes secrets**
+           - A list of [Kubernetes Secrets](https://kubernetes.io/docs/concepts/configuration/secret/) to be accessed as environment variables during Jupyter notebook or script execution. Format: `ENV_VAR=secret-name:secret-key`. Entries that are empty (`ENV_VAR=`) or malformed are ignored. Entries with a secret name considered to be an [invalid Kubernetes resource name](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names) or with [an invalid secret key](https://kubernetes.io/docs/concepts/configuration/secret/#restriction-names-data) will raise a validation error after pipeline submission or export.
+           - The referenced secrets must exist in the Kubernetes namespace where the generic pipeline nodes are executed.
+           - Secrets are ignored when the pipeline is executed locally. For remote execution, in the case where a node has a Kubernetes secret and an environment variable with the same name, the Kubernetes secret will be preferred and the corresponding environment variable will be removed from the list.
```

Comment:
Two nits:
 - The wording is a bit ambiguous 'has a Kubernetes secret and an environment variable with the same name': should try to clarify that 'same name' refers to the environment variable name that the secret is associated with
 - The wording 'the corresponding environment variable will be removed from the list' unintentionally seems to target developers not users because it references internal behavior. 

Perhaps change this to something like

For remote execution, if an environment variable was assigned a static value (via the `environment variables` property) and a secret's value, the secret's value is used.

An example might also help.